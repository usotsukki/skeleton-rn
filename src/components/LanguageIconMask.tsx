import Entypo from '@expo/vector-icons/Entypo'
import MaskedView from '@react-native-masked-view/masked-view'
import React from 'react'
import { StyleSheet } from 'react-native'
import { Image, View } from 'react-native-ui-lib'

const LanguageIconMask = () => {
	return (
		<MaskedView
			testID={'language-icon.container'}
			style={styles.maskContainer}
			maskElement={
				<View testID={'language-icon.mask'} style={styles.maskElement}>
					<Entypo name="language" size={300} />
				</View>
			}>
			<Image testID={'language-icon.image'} width={700} assetGroup="illustrations" assetName="graphicBackground" />
		</MaskedView>
	)
}

const styles = StyleSheet.create({
	maskContainer: { flex: 1, flexDirection: 'row', height: '100%', justifyContent: 'center', alignItems: 'center' },
	maskElement: {
		flex: 1,
		backgroundColor: 'transparent',
		justifyContent: 'center',
		alignItems: 'center',
	},
})

export default LanguageIconMask
