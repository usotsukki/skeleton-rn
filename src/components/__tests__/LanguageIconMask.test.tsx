import { render, screen } from '@testing-library/react-native'
import React from 'react'
import LanguageIconMask from '../LanguageIconMask'

describe('LanguageIconMask', () => {
	afterEach(() => {
		jest.clearAllMocks()
	})

	it('renders correctly', () => {
		const renderTree = render(<LanguageIconMask />)

		expect(renderTree).not.toBeNull()
		expect(screen.getByTestId('language-icon.container')).toBeTruthy()
		expect(screen.getByTestId('language-icon.mask')).toBeTruthy()
		expect(screen.getByTestId('language-icon.image')).toBeTruthy()
		expect(renderTree.toJSON()).toMatchSnapshot()
	})
})
