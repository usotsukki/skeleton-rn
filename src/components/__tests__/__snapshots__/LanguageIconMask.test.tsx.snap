// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LanguageIconMask renders correctly 1`] = `
<RNCMaskedView
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "row",
      "height": "100%",
      "justifyContent": "center",
    }
  }
  testID="language-icon.container"
>
  <View
    pointerEvents="none"
    style={
      {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <View
      style={
        [
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "flex": 1,
            "justifyContent": "center",
          },
        ]
      }
      testID="language-icon.mask"
    >
      <Text />
    </View>
  </View>
  <Image
    accessibilityRole="image"
    accessible={false}
    assetGroup="illustrations"
    assetName="graphicBackground"
    onError={[Function]}
    style={
      [
        undefined,
        undefined,
        {
          "width": 700,
        },
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        false,
        undefined,
        false,
      ]
    }
    testID="language-icon.image"
  />
</RNCMaskedView>
`;
